src/main/java/com/sforce/cd/apexUnit/client/connection/ConnectionHandler.java

```diff
@@ -217,13 +229,19 @@ public void setBulkConnection(BulkConnection bulkConnection) {
 		this.bulkConnection = bulkConnection;
 	}
 
-	public static String logConnectionException(ConnectionException connEx) {
-		return "Exception thrown while trying to create Partner Connection!!!" + connEx.getMessage();
+	public static String logConnectionException(ConnectionException connEx, PartnerConnection conn) {
+		return logConnectionException(connEx, conn, null);
 	}
 
-	public static String logConnectionException(ConnectionException e, String soql) {
-		return "Connection Exception encountered when trying to query : " + soql
-				+ " \n The connection exception description says : " + e.getMessage();
+	public static String logConnectionException(ConnectionException connEx, PartnerConnection conn, String soql) {
+		StringBuffer returnString = new StringBuffer("Connection Exception encountered ");
+		if (null != soql) {
+			returnString.append("when trying to query : " + soql);
+		}
+		returnString.append(" The connection exception description says : " + connEx.getMessage());
+		returnString.append(" Object dump for the Connection object: " + ReflectionToStringBuilder.toString(conn));
```

Comment:
Please verify that this object is in a readable format. Otherwise you may have to override toString()